microsoft/vscode
# Allow FileSystemProvider to stat a file as readonly (#73122)

This draft PR addresses #73122

It adds an optional `readonly` boolean property to the FileStat interface, thus enabling a FileSystemProvider to control whether a file is editable or read-only.

One way to test:
1. Clone https://github.com/gjsjohnmurray/vscode-extension-samples/tree/test-vscode-73122-fsp-stat-readonly
2. Open fsprovider-sample folder.
3. Add this PR's vscode.proposed.d.ts file to that folder.
4. Build and run.
5. Run the commands to init the MemFS workspace and add the MemFS files.
6. Open any of the .txt files to see them read-only.
7. Open any of the other files to see that they are editable as usual.

Submitted as a draft PR seeking comment and guidance from @jrieken about taking this through the API proposal precedure.
